openedx-unsupported/configuration
# feature: Allow installing extra requirements when using the `credentials` role

This makes the `credentials` role support the use of `edx_django_service_extra_requirements`
as done originally in [!6400](https://github.com/edx/configuration/pull/6400)

The end-goal of this is to support theming of the credentials service as described [here](https://edx-credentials.readthedocs.io/en/latest/theming.html#installing-a-theme)

Related tickets:
* [BB-4342](https://tasks.opencraft.com/browse/BB-4342)

# Testing

1. Launch devstack with credentials service: `make dev.up.credentials`
2. In the `configuration` repo, create a `docker_inventory.ini` file containing just
```ini
credentials ansible_connection=docker ansible_host=edx.devstack.credentials
```
3. Create a playbook `credentials_test.yml`:
```yml
- name: Deploy edX Credentials Service
  hosts: all
  become: True
  gather_facts: True
  vars:
    ENABLE_DATADOG: False
    ENABLE_NEWRELIC: False
    CLUSTER_NAME: 'credentials'
  roles:
    - credentials
```
4. Run the playbook from the `playbooks` directory (to ensure ansible.cfg is picked up): `ansible-playbook credentials_test.yml -i docker_inventory.ini --skip-tags install:code -e '{ disable_edx_services: true, CREDENTIALS_EXTRA_REQUIREMENTS: [{name: 'lolcat'}], edx_django_service_is_devstack: true}'`
5. Assert `lolcat` is now installed in the credentials virtualenv:
```sh
make dev.shell.credentials
source /edx/app/credentials/venvs/credentials/bin/activate
echo test | lolcat 
```
